src/renderer/renderScene.ts

```diff
@@ -314,482 +319,559 @@ const renderLinearElementPointHighlight = (
   context.restore();
 };
 
-export const _renderScene = ({
+export const renderCanvasContent = ({
   elements,
   appState,
   scale,
   rc,
   canvas,
-  renderConfig,
+  canvasContentRenderConfig,
 }: {
   elements: readonly NonDeletedExcalidrawElement[];
   appState: AppState;
   scale: number;
   rc: RoughCanvas;
   canvas: HTMLCanvasElement;
-  renderConfig: RenderConfig;
-}) =>
-  // extra options passed to the renderer
-  {
-    if (canvas === null) {
-      return { atLeastOneVisibleElement: false };
-    }
-    const {
-      renderScrollbars = true,
-      renderSelection = true,
-      renderGrid = true,
-      isExporting,
-    } = renderConfig;
-
-    const selectionColor = renderConfig.selectionColor || oc.black;
+  canvasContentRenderConfig: CanvasContentRenderConfig;
```

Comment:
FWIW we can keep the param name as `renderConfig`, just differentiate the type.

Reply:
done